### PR TITLE
Refactor the classes

### PR DESCRIPTION
**Pull request by Sirvelasque**
- changed the book declaration
- changed the add and remove functions to methods
- styled

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,70 +1,98 @@
...
+}
 // call nedded elements from the document
 const bookList = document.querySelector('.book-list');
 const form = document.getElementById('abbBookForm');
 const title = document.getElementById('title');
 const author = document.getElementById('author');
+
 // create the books, add them the book list and local storage
 form.addEventListener('submit', (e) => {
   e.preventDefault();
```
Comment (andersonlebon) [inline main.js]: - [x] One of the requirements says that you should refactor your code using classes, which means all logic regarding the book logic like store, adding a new book, removing a book.... should be methods inside your class. for example
```javascript
class Books {
  constructor() {
    this.books = [];
  }

  addBook(title, author) {
    id = generateID(); 
    this.books = this.books.concat({ title, author, author });
    return this.books; 
  }

  removeBook(bookID) {
    // bookID can be any other property, e.g. title or author, in as much is it unique
    this.books = this.books.filter(({ id }) => bookID !== id); // This assumes that `bookID` parameter is the book id
    return this.books; 
  }

saveBookInLocalstorage(book){
     .........
}

  // getter method
  getBooks() {
    return this.books;
  }
}

```

- [x] Kindly work also on this error when in the store there are books with the same title, deleting one also deletes the other of the same title and is not removed in the browser. Kindly check this :point_down:  screenvideo for more clarification

https://user-images.githubusercontent.com/65068771/154213508-d813b63d-7943-47de-b797-250631366657.mp4

**Review by andersonlebon — COMMENTED** [review]
## Hi team @eng-mohamed-eldeeb  && @eng-mohamed-eldeeb  

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] I would strongly recommend add a section in your README that gives instruction on how to get and run you project locally.
image


- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js [hunk 2]
```diff
@@ -1,70 +1,90 @@
...
+}
+
+function deleteBook(i) {
+  let currentBooks = localStorage.getItem('books');
+  currentBooks = JSON.parse(currentBooks);
+  currentBooks.splice(i, 1);
+  currentBooks = JSON.stringify(currentBooks);
+  localStorage.setItem('books', currentBooks);
+  show();
+}
```
Comment (julie-ify) [inline main.js]: - [ ] As requested by the previous reviewer, kindly add this function as a method in the `Book` class as it is a requirement for this project.

File: main.js [hunk 3]
```diff
@@ -1,70 +1,90 @@
...
+      pForDelete.textContent = 'Remove';
+      // add element to the list and the list to the ul
+      liP2.appendChild(pForTitle);
+      liP1.appendChild(pForAuthor);
+      ulPContaner.innerHTML = `"${book.title}" by ${book.author}`;
+      // ulPContaner.appendChild(liP1);
+      // ulPContaner.appendChild(liP2);
+      li.appendChild(ulPContaner);
+      // li.appendChild(pForTitle);
+      // li.appendChild(pForAuthor);
```
Comment (julie-ify) [inline main.js]: - [x] Kindly remove these commented lines of code as it is not a good practice to have unnecessary commented lines of code in the codebase.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @eng-mohamed-eldeeb and @Sirvelasque, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hello, team,

You've done well working on the concerns from the previous reviews 👍🏿.
Tasks for this milestone are now successfully implemented 💯.
All the best with the next milestone.

You are free to merge this branch :shipit:

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**